RpcParser/markdown_generator.py

```diff
@@ -0,0 +1,102 @@
+"""Generator for Markdown
+
+"""
+import re
+import sys
+from pathlib import Path
+
+sys.path.append(Path(__file__).absolute().parents[1].joinpath('InterfaceParser').as_posix())
+try:
+    from model.function import Function
+    from model.enum import Enum
+    from model.struct import Struct
+    from model.enum_subset import EnumSubset
+    from model.array import Array
+    from model.double import Double
+
+    from parsers.rpc_base import ParseError
+    from parsers.sdl_rpc_v2 import Parser
+except ModuleNotFoundError as error:
+    print(error)
+    sys.exit(1)
+
+
+def extract_type(param):
+    """Evaluate and extract type
+    :param param: sub-element (Param, FunctionParam) of element from initial Model
+    :return: string with sub-element type
+    """
+
+    def evaluate(kind):
+        if isinstance(kind, EnumSubset):
+            return kind.enum.name
+        if isinstance(kind, (Struct, Enum)):
+            return kind.name
+        if isinstance(kind, Double):
+            return 'Float'
+        return type(kind).__name__
+
+    if isinstance(param.param_type, Array):
+        return '{}[]'.format(evaluate(param.param_type.element_type))
+    return evaluate(param.param_type)
+
+
+def main(readme_file='../README.md', file_xml='../MOBILE_API.xml'):
```

Comment:
I believe the generator is supposed to take a number of command line arguments as described [in the proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0234-proxy-rpc-generation.md#command-line-switches)?

Reply:
@JackLivio based on [the review discussion](https://github.com/smartdevicelink/sdl_evolution/issues/853), there is mentioned that README.md generator should include `rpc_spec` scripts. The same correct for the `InterfaceGenerator` in the `sdl_core`, the PR will be provided shortly. The main goal on our part before doing any changes in the old code was to do changes with care leaving the existing functionality work, otherwise, by putting the new command-line switches, it requires deeper refactoring and improvements of the existing code. As both old markdown generator and `InterfaceGenerator` in the `sdl_core` are refactored to include `rpc_spec` scripts, in my opinion, we follow the proposal requirements regarding those parts and the command-line switches are strict for new generators in libraries. What do you think?

Reply:
Sorry to quote the proposal but:  `The command-line switches of all generator scripts should be identical.`

I believe that this markdown generator and the core generator using this repositories parser should have identical command line switches as the other projects.

Reply:
currently markdown_generator.py supports all arguments mentioned here:
https://github.com/smartdevicelink/sdl_evolution/issues/853#issuecomment-548346897